src/components/TeamMemberTasks/ReviewButton.jsx

```diff
@@ -0,0 +1,125 @@
+import React, { useState } from 'react';
+import { Button, Modal, ModalHeader, ModalBody, ModalFooter, DropdownToggle, DropdownMenu, DropdownItem, UncontrolledDropdown } from 'reactstrap';
+import './style.css';
+import './reviewButton.css';
+import { boxStyle } from 'styles';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCheck } from '@fortawesome/free-solid-svg-icons';
+
+const ReviewButton = ({
+  user,
+  myUserId,
+  myRole,
+  task,
+  updateTask
+}) => {
+
+  const [modal, setModal] = useState(false);
+
+  const toggleModal = () => {
+    setModal(!modal);
+  };
+  
+  const reviewStatus = function() {
+    let status = "Unsubmitted";
+    for(let resource of task.resources){
+      if (resource.userID === user.personId) {
+        status = resource.reviewStatus ? resource.reviewStatus : "Unsubmitted"
+        break;
+      }
+    }
+    return status;
+  }();
+
+  const updReviewStat = (newStatus) => {
+    const resources = [...task.resources];
+    const newResources = resources?.map(resource => {
+      let newResource = { ...resource };
+      if (resource.userID === user.personId) {
+        newResource = newStatus === "Reviewed"
+        ? {
+          ...resource,
+          completedTask: true,
+          reviewStatus: newStatus,
+        } : {
+          ...resource,
+          reviewStatus: newStatus,
+        };
+      }
+      return newResource;
```

Comment:
Would this do the same as you intended?
```
const newResource = { ...resource };
if (resource.userID === user.personId) {
  newResources.reviewStatus = newStatus;
  newResources.completedTask = newStatus === "Reviewed";
}
return newResource;
```

Reply:
updated the code based on your above three suggestions, please check.